File: arrays-hashing/09-longest-consecutive-sequence/main_test.py
import unittest

from main import Solution


class TestLongestConsecutive(unittest.TestCase):
    def test_example_1(self):
        result = Solution().longestConsecutive([2, 20, 4, 10, 3, 4, 5])
        self.assertEqual(result, 4)

    def test_example_2(self):
        result = Solution().longestConsecutive([0, 3, 2, 5, 4, 6, 1, 1])
        self.assertEqual(result, 7)


if __name__ == "__main__":
    unittest.main()

File: two-pointers/05-trapping-rain-water/MainTest.java
public class MainTest {
    static int passed = 0;
    static int failed = 0;

    public static void main(String[] args) {
        check("example 1", () -> {
            int result = new Solution().trap(new int[]{0, 2, 0, 3, 1, 0, 1, 3, 2, 1});
            assertEqual(9, result);
        });

        check("example 2", () -> {
            int result = new Solution().trap(new int[]{4, 2, 0, 3, 2, 5});
            assertEqual(9, result);
        });

        System.out.println(passed + " passed, " + failed + " failed");
        if (failed > 0) {
            System.exit(1);
        }
    }

    interface TestCase {
        void run() throws Exception;
    }

    static void check(String name, TestCase test) {
        try {
            test.run();
            passed++;
            System.out.println("PASS: " + name);
        } catch (Throwable e) {
            failed++;
            System.out.println("FAIL: " + name + " -> " + e.getMessage());
        }
    }

    static void assertEqual(int expected, int actual) {
        if (expected != actual) {
            throw new AssertionError("expected " + expected + " but got " + actual);
        }
    }
}
